easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,353 @@
+        """Custom easyconfig parameters for LAMMPS"""
+
+        extra_vars = {
+            # see https://developer.nvidia.com/cuda-gpus
+            'cuda_compute_capabilities': [[], "List of CUDA compute capabilities to build with", CUSTOM],
+            'general_packages': [None, "List of general packages without `PKG_` prefix.", MANDATORY],
+            'kokkos': [True, "Enable kokkos build. (enabled by default)", CUSTOM],
+            'kokkos_arch': [None, "Set kokkos processor arch manually, if auto-detection doesn't work.", CUSTOM],
+            'user_packages': [None, "List user packages without `PKG_USER-` prefix.", MANDATORY],
+        }
+        return CMakeMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration procedure for LAMMPS."""
+
+        cuda = get_software_root('CUDA')
+        # list of CUDA compute capabilities to use can be specifed in two ways (where (2) overrules (1)):
+        # (1) in the easyconfig file, via the custom cuda_compute_capabilities;
+        # (2) in the EasyBuild configuration, via --cuda-compute-capabilities configuration option;
+        ec_cuda_cc = self.cfg['cuda_compute_capabilities']
+        cfg_cuda_cc = build_option('cuda_compute_capabilities')
+        cuda_cc = cfg_cuda_cc or ec_cuda_cc or []
+
+        # cmake has its own folder
+        self.cfg['srcdir'] = os.path.join(self.start_dir, 'cmake')
+
+        # verbose CMake
+        if '-DCMAKE_VERBOSE_MAKEFILE=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DCMAKE_VERBOSE_MAKEFILE=yes')
+
+        # Enable following packages, if not configured in easycofig
+        default_options = [
+            'BUILD_DOC', 'BUILD_EXE', 'BUILD_LIB',
+            'BUILD_SHARED_LIBS', 'BUILD_TOOLS',
+        ]
+        for option in default_options:
+            if "-D%s=" % option not in self.cfg['configopts']:
+                self.cfg.update('configopts', '-D%s=on' % option)
+
+        # Is there a gzip
+        if '-DWITH_GZIP=' not in self.cfg['configopts']:
+            if get_software_root('gzip'):
+                self.cfg.update('configopts', '-DWITH_GZIP=yes')
+            else:
+                self.cfg.update('configopts', '-DWITH_GZIP=no')
+
+        # Is there a libpng
+        if '-DWITH_PNG=' not in self.cfg['configopts']:
+            if get_software_root('libpng'):
+                self.cfg.update('configopts', '-DWITH_PNG=yes')
+            else:
+                self.cfg.update('configopts', '-DWITH_PNG=no')
+
+        # Is there a libjpeg-turbo
+        if '-DWITH_JPEG=' not in self.cfg['configopts']:
+            if get_software_root('libjpeg-turbo'):
+                self.cfg.update('configopts', '-DWITH_JPEG=yes')
+            else:
+                self.cfg.update('configopts', '-DWITH_JPEG=no')
+
+        # With Eigen dependency:
+        if '-DDOWNLOAD_EIGEN3=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DDOWNLOAD_EIGEN3=no')
```

Comment:
I think this should stay. If I remember correctly the LAMMPS was failing to build with self-downloaded Eigen probably because of wrong or duplicated paths.

I don't think we should auto-download stuff, if something (cmake?) determines it is missing Eigen just because it will not find it? Especially if there is Easyblock for Eigen present, and you can override downloding settings from easyconfig - so if you know what you are doing, you can let lammps download eigen automatically.

Reply:
That part of the code should complaint loudly if Eigen is specified as a dependency in the easyconfig and the DOWNLOAD_EIGEN option is also specified.